component/src/main/java/com/celements/structEditor/DefaultStructuredDataEditorService.java

```diff
@@ -286,26 +276,21 @@ public Optional<String> getRequestOrCellValue(XWikiDocument cellDoc, XWikiDocume
 
   @Override
   public Optional<Date> getCellDateValue(XWikiDocument cellDoc, XWikiDocument onDoc) {
-    Object value = getCellValue(cellDoc, onDoc);
-    if (value instanceof Date) {
-      return Optional.of((Date) value);
-    }
-    return Optional.empty();
+    return getCellValue(cellDoc, onDoc)
+        .flatMap(tryCastOpt(Date.class));
   }
 
   @Override
   public List<String> getCellListValue(XWikiDocument cellDoc, XWikiDocument onDoc) {
-    List<String> ret = new ArrayList<>();
-    Object value = getCellValue(cellDoc, onDoc);
-    if (value instanceof List) {
-      for (Object elem : (List<?>) value) {
-        ret.add(elem != null ? elem.toString() : "");
-      }
-    }
-    return ret;
+    return getCellValue(cellDoc, onDoc)
+        .flatMap(tryCastOpt(List.class))
+        .stream().<Object>flatMap(List::stream)
+        .map(elem -> (elem != null) ? elem.toString() : "")
+        .collect(toList());
```

Comment:
Jenkins doesn't build.

[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /var/jenkins_home/jobs/celements-struct (jar)/workspace/component/src/main/java/com/celements/structEditor/DefaultStructuredDataEditorService.java:[289,17] incompatible types: java.lang.Object cannot be converted to java.util.List<java.lang.String>